device/tt_device.h

```diff
@@ -232,170 +234,216 @@ class tt_device
     }
 
     /**
-    * @brief Set Host Address Map parameters used by UMD to communicate with the TT Device (used for remote transactions)
-    * \param host_address_params_ tt_driver_host_address_params encapsulating all the Host Address space parameters required by UMD
-    */ 
-
+     * Set Host Address Map parameters used by UMD to communicate with the TT Device (used for remote transactions).
+     *
+     * | Argument             | Description                                             | Type                           | Valid Range | Required |
+     * |----------------------|---------------------------------------------------------|--------------------------------|-------------|----------|
+     * | host_address_params_ |  All the Host Address space parameters required by UMD. | tt_driver_host_address_params_ |             | Yes      |
+     */ 
     virtual void set_driver_host_address_params(const tt_driver_host_address_params& host_address_params_) {
         throw std::runtime_error("---- tt_device::set_driver_host_address_params is not implemented\n");
     }
 
     /**
-    * @brief Set ERISC Firmware parameters used by UMD to communicate with the TT Device (used for remote transactions)
-    * \param eth_interface_params_ tt_driver_eth_interface_params encapsulating all the Ethernet Firmware parameters required by UMD
-    */ 
+     * Set ERISC Firmware parameters used by UMD to communicate with the TT Device (used for remote transactions).
+     *
+     * | Argument              | Description                                           | Type                            | Valid Range | Required |
+     * |-----------------------|-------------------------------------------------------|---------------------------------|-------------|----------|
+     * | eth_interface_params_ | All the Ethernet Firmware parameters required by UMD. | tt_driver_eth_interface_params_ |             | Yes      |
+     */ 
     virtual void set_driver_eth_interface_params(const tt_driver_eth_interface_params& eth_interface_params_) {
         throw std::runtime_error("---- tt_device::set_driver_eth_interface_params is not implemented\n");
     }
 
     /**
-    * @brief Configure a TLB to point to a specific core and an address within that core. Should be done for Static TLBs
-    * \param logical_device_id Logical Device being targeted
-    * \param core The TLB will be programmed to point to this core
-    * \param tlb_index TLB id that will be programmed
-    * \param address All incoming transactions to the TLB will be routed to an address space starting at this parameter (after its aligned to the TLB size)
-    * \param ordering Ordering mode for the TLB. Can be Strict (ordered and blocking, since this waits for ack -> slow), Relaxed (ordered, but non blocking -> fast) or Posted (no ordering, non blocking -> fastest).
-    */ 
+     * Configure a TLB to point to a specific core and an address within that core. Should be done for Static TLBs.
```

Comment:
Is this table some standard Doxygen way? Or is this how it is documented in tt_metal, and actually a preparation for sphinx?

I also wonder what would be the right way to have longer argument comments, this format caps them inconveniently.

Reply:
Agree w/ @broskoTT.  The table format looks like it has a higher modification effort than e.g.
```
/**
 * @param p <some description> 
 * \param z <some description> (not sure if \ or @ is more Doxygen-y)
 */
```
and also seems like it will increase line length past 120 chars.

Reply:
I was just following how tt-metal does it. I think it would be nice to have it in the same way, but don't really have strong opinions on it. Doxygen is ok with both formats that are specified, it is just a matter of what we choose. I will do whatever we agree on.

Just to point out, the format Joel specifed was being used, but it seemed very messy to me. It does give people more freedom to describe the param, but if too much text is needed to explain some param, maybe the API is broken or param is too complicated

As long as we put in the docs which format we want to follow I think we are good.
Maybe we can ask tt-metal folks as well if they want anything specific on the format

Reply:
As per offline discussion, resolving comment, keeping compatibility with tt-metal for now